### DISCUSSION TITLE
Compilation Config fails running a ninja subcommand

### DISCUSSION THREAD

**Discussion by holly564** [musescore/MuseScore#17288] [General]
Compiled the master branch on Windows 10.
Current environment: [MSVC 2019]  [Qt_5_15_2] [Qt Creator 10 (Community) ] [cmake 3.24.2(Qt) installed but using [cmake version 3.20.0]] [ninja as cmake_generator]
REM: ninja path has been added to the list of environment vars.

Followed each step of https://github.com/musescore/MuseScore/wiki/Compile-in-Qt-Creator but the compilation config fails the same way if I ran the compilation cmd via CLI
CLI environment setup: [cmake version 3.20.0]
[cygwin

Can't find out what I missed identifying in the lines below:
QtCreator Output Console:
`
Running C:\cygwin64\bin\cmake.exe 
-S C:/Users/Holly/Projects/MuseScore 
-B C:/Users/Holly/Projects/MuseScore/builds/Desktop_Qt_5_15_2_MSVC2019_64bit-Debug "-DCMAKE_GENERATOR:STRING=Ninja" 
"-DCMAKE_BUILD_TYPE:STRING=Debug" 
"-DCMAKE_PROJECT_INCLUDE_BEFORE:FILEPATH=C:\Users\Holly\Projects\MuseScore\builds\Desktop_Qt_5_15_2_MSVC2019_64bit-Debug/.qtc/package-manager/auto-setup.cmake" 
"-DQT_QMAKE_EXECUTABLE:FILEPATH=C:/Qt/5.15.2/msvc2019_64/bin/qmake.exe" "-DCMAKE_PREFIX_PATH:PATH=C:/Qt/5.15.2/msvc2019_64" 
"-DCMAKE_C_COMPILER:FILEPATH=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe" 
"-DCMAKE_CXX_COMPILER:FILEPATH=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe" 
"-DCMAKE_INSTALL_PREFIX:FILEPATH=install" 
"-DCMAKE_CXX_FLAGS_INIT:STRING=-DQT_QML_DEBUG" in C:\Users\Holly\Projects\MuseScore\builds\Desktop_Qt_5_15_2_MSVC2019_64bit-Debug.
-- The C compiler identification is MSVC 19.29.30148.0
-- The CXX compiler identification is MSVC 19.29.30148.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe - broken
CMake Error at /usr/share/cmake-3.20.0/Modules/CMakeTestCCompiler.cmake:66 (message):
  The C compiler

    "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/Users/Holly/Projects/MuseScore/builds/Desktop_Qt_5_15_2_MSVC2019_64bit-Debug/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/cygdrive/c/Users/Holly/AppData/Local/Programs/Python/Python38/Scripts/ninja.exe cmTC_6adee && [1/2] Building C object CMakeFiles/cmTC_6adee.dir/testCCompiler.c.o
    Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30148 for x64

    Copyright (C) Microsoft Corporation.  All rights reserved.
    cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release

    testCCompiler.c

    [2/2] Linking C executable cmTC_6adee.exe
    FAILED: cmTC_6adee.exe 
    : && "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe"   CMakeFiles/cmTC_6adee.dir/testCCompiler.c.o -o cmTC_6adee.exe   && :
    CreateProcess failed: The system cannot find the file specified.
    ninja: build stopped: subcommand failed.
    
  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:25 (project)

-- Configuring incomplete, errors occurred!
See also "C:/Users/Holly/Projects/MuseScore/builds/Desktop_Qt_5_15_2_MSVC2019_64bit-Debug/CMakeFiles/CMakeOutput.log".
See also "C:/Users/Holly/Projects/MuseScore/builds/Desktop_Qt_5_15_2_MSVC2019_64bit-Debug/CMakeFiles/CMakeError.log".
CMake process exited with exit code 1.

Elapsed time: 00:14.
`

Thanks for any tip/remark

**Comment by cbjeukendrup** [comment]
There is a problem with the compiler. CMake thinks that it is broken. Maybe that is true; in that case, reinstalling Visual Studio and all the tools might fix it. 

I also wonder whether it might be Cygwin what's causing trouble. Maybe that confuses CMake and makes it treat the MSVC compiler as some other kind of compiler. But probably it makes no sense what I'm saying there; I hardly know what Cygwin is (and I'm not even a Windows user...)

**Reply by holly564** [reply]
Morning,

Deleted cygwin and noticed that my cmake-generator variable initialized to "Visual Studio"  was incorrectly set
Updated that and the configuration ran smoothly.

Thank you

**Reply by cbjeukendrup** [reply]
Glad to hear that!
